common/Reducers.jsm

```diff
@@ -23,7 +23,14 @@ const INITIAL_STATE = {
     // The history (and possibly default) links
     rows: [],
     // Used in content only to dispatch action to TopSiteForm.
-    editForm: null
+    editForm: null,
+    // Used in content only to open the SearchShortcutsForm modal.
+    showSearchShortcutsForm: false,
+    // The list of available search shortcuts.
+    searchShortcuts: [
+      {keyword: "@google", shortURL: "google", url: "https://www.google.com", icon: "resource://activity-stream/data/content/tippytop/images/google-com@2x.png"},
+      {keyword: "@amazon", shortURL: "amazon", url: "https://www.amazon.com", icon: "resource://activity-stream/data/content/tippytop/images/amazon@2x.png"}
```

Comment:
This needs to be hooked up with real data from somewhere. @Mardak ?

Reply:
@sarracini @AdamHillier I would think this data needs to match the format of https://github.com/mozilla/activity-stream/blob/9ddc1713fab5dfbc7f90321ad6c4c69d38535b3e/lib/SearchShortcuts.jsm#L9-L12

We'll want to derive the data from something like `Services.search.getEngines().map(v => v.wrappedJSObject._internalAliases)` from https://hg.mozilla.org/integration/autoland/rev/1f187635f2e4#l3.12

Reply:
We can add this to the reducer from `TopSitesFeed`

Reply:
Notably, the urls used here are different from those in SearchShortcuts, which will affect blocking